intelmq/bots/outputs/redis/output.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+import sys
+import time
+import redis
+import intelmq.lib.utils as utils
+from intelmq.lib.bot import Bot
+
+
+class RedisOutputBot(Bot):
+
+    def init(self):
+        self.host = self.parameters.redis_server_ip
+        self.port = int(self.parameters.redis_server_port)
+        self.db = int(self.parameters.redis_db)
+        self.queue = self.parameters.redis_queue
+        self.password = self.parameters.redis_password
+        self.timeout = int(self.parameters.redis_timeout)
+
+        self.conn = redis.ConnectionPool(host=self.host, port=self.port, db=self.db)
+        self.connect()
+
+    def process(self):
+        event = self.receive_message()
+        
+        if event is None:
+           self.acknowledge_message()
+           return
+
+        while True:
+              try:
+                  self.output.lpush(self.queue, utils.encode(event.to_json()))
+                  break
+              except redis.RedisError as e:
+                  self.logger.error('Redis: failled to sent message!')
+                  self.logger.error(e)
+                  self.connect()
+
+        self.acknowledge_message()
+    
+    def connect(self):
+        while True:
```

Comment:
Again, please remove the loop and the sleep. The bot is allowed to fail, we do have proper error-handling an sleeping. See https://github.com/certtools/intelmq/blob/master/intelmq/lib/bot.py#L82
As I said, adapting the default values for retry-times etc. for this bot is allowed and you can give recommendations in BOTS